nap/utils.py

```diff
@@ -1,9 +1,24 @@
 from __future__ import unicode_literals
 import itertools
+from urllib import urlencode
+from urlparse import parse_qsl, urlsplit, urlunsplit
+
 import six
 from six.moves import urllib
 
 
+def normalize_url(url_string):
+    url = urlsplit(url_string)
+    if url.query:
+        query_params = sorted(parse_qsl(url.query, keep_blank_values=True), key=lambda key_value: key_value[0])
+        query_normalized = urlencode(query_params)
+        url = url._replace(query=query_normalized)
```

Comment:
I dislike calling a private method like this. Could you not just add the query part back as a parameter to urlunsplit?